### PR TITLE
Better frame delay precision

### PR DESCRIPTION
**Pull request by AJenbo**
- Most refresh rates aliging poorly with integer ms
- Adaptive frame limit for low spec devices

### CODE REVIEW

File: SourceX/dx.cpp [hunk 1]
```diff
@@ -205,6 +205,21 @@ void BltFast(DWORD dwX, DWORD dwY, LPRECT lpSrcRect)
 	bufferUpdated = true;
 }
 
+/**
+ * @brief Limit FPS to lower CPU when not waiting for V-Blank
```
Comment (glebm) [inline SourceX/dx.cpp]: s/not waiting/waiting/?

File: SourceX/miniwin/ddraw.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 namespace dvl {
 
-extern int refreshDelay;
+extern double refreshDelay;
```
Comment (glebm) [inline SourceX/miniwin/ddraw.h]: Instead of using floating point math, consider changing the units to nanoseconds

File: SourceX/dx.cpp [hunk 3]
```diff
@@ -205,6 +205,21 @@ void BltFast(DWORD dwX, DWORD dwY, LPRECT lpSrcRect)
 	bufferUpdated = true;
 }
 
+/**
+ * @brief Limit FPS to lower CPU when not waiting for V-Blank
+ */
+void LimitFrameRate()
+{
+	static uint32_t frameDeadline;
```
Comment (glebm) [inline SourceX/dx.cpp]: nit: These should be `std::uint32_t` and `std::fmod` in C++

File: SourceX/dx.cpp (hunk 1 above)
Comment (AJenbo) [reply SourceX/dx.cpp]: No, this function is to be used when *waiting for v-blank* (aka v-sync) is not avalible.

File: SourceX/dx.cpp (hunk 3 above)
Comment (AJenbo) [reply SourceX/dx.cpp]: This gives me a build error saying uint32_t isn't part of std

**Comment by AJenbo** [comment]
Testing on RG300 the adaptive delay turns out to be a huge improvement for low spec devices. FPS in town jumped from 20-21 to 30-31, and the dungeon FPS went from 16-18 to 19-21.

File: SourceX/miniwin/ddraw.h (hunk 2 above)
Comment (AJenbo) [reply SourceX/miniwin/ddraw.h]: Done

File: SourceX/dx.cpp (hunk 3 above)
Comment (glebm) [reply SourceX/dx.cpp]: `#include <cstdint>`

**Comment by glebm** [comment]
Woo-hoo!

**Pull request merged**